umh-core/pkg/service/topicbrowser/parse.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2025 UMH Systems GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package topicbrowser
+
+import (
+	"bytes"
+	"encoding/hex"
+	"errors"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pierrec/lz4/v4"
+	s6svc "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/service/s6"
+)
+
+// parseBlock parses the logs, decompresses the payload and sets the timestamp
```

Comment:
can you elaborate here more? it takes in new entries, and will updat teh svc.ringbyffer with the new entry

Reply:
lso write down the assumptions here. that you will get a hex encoded blob in what format